From 129cad165d876405eb6c220e63452e4414cc2f7e Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 24 Oct 2018 10:49:13 +0200
Subject: [PATCH] arm64: dt: fsl-imx8qxp-tqma8qx-mba8qx: add usb otg 1

- baseboard V5V regulator
- pin mux for OC / PWR
- host config, disable OC detection

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 .../freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi
index 7804523484d4..add1010e4758 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-tqma8qx-mba8qx.dtsi
@@ -38,6 +38,14 @@
 			startup-delay-us = <100>;
 			off-on-delay-us = <12000>;
 		};
+
+		reg_mba8qx_v5v: mba8qx_v5v {
+			compatible = "regulator-fixed";
+			regulator-name = "MBA8QX_V5V";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			regulator-always-on;
+		};
 	};
 };
 
@@ -99,6 +107,13 @@
 			>;
 		};
 
+		pinctrl_usbotg1: usbotg1grp {
+			fsl,pins = <
+				SC_P_USB_SS3_TC0_CONN_USB_OTG1_PWR	0x06000020
+				SC_P_USB_SS3_TC2_CONN_USB_OTG1_OC	0x06000020
+			>;
+		};
+
 		pinctrl_usdhc2_gpio: usdhc2gpiogrp {
 			fsl,pins = <
 				SC_P_USDHC1_RESET_B_LSIO_GPIO4_IO19	0x06000021
@@ -203,6 +218,17 @@
 	status = "okay";
 };
 
+&usbotg1 {
+	vbus-supply = <&reg_mba8qx_v5v>;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	power-polarity-active-high;
+	disable-over-current;
+	dr_mode = "host";
+	status = "okay";
+};
+
 &usdhc2 {
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
